.github/workflows/pr-validation.yml

```diff
@@ -0,0 +1,146 @@
+# For more information see:
+# - https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+# - https://docs.github.com/en/actions/learn-github-actions/contexts#github-context
+
+name: pr-validation
+
+on:
+  pull_request:
+
+defaults:
+  run:
+    shell: bash
+
+concurrency:
+  group: ${{ github.head_ref }}.${{ github.sha }}.pr-validation
+  cancel-in-progress: true
+
+jobs:
+  premerge-build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
```

Comment:
@markwhitfeld `fetch-depth: 0` stands for all history for all branches and tags. It should be a bit slower and will pull much more data than needed for other workflows than those that validate pull requests. From my experience with common repos and monorepos, it is generally not needed to pull all git history in the `trunk` or `release` workflows, because all tests that might require full commit history of all branches and tags (e.g. validate commit messages using commitlint) were done earlier during the corresponding PR validation.

If you think the default `fetch-depth: 0` makes sense, so be it.

Reply:
Ok. Makes sense. Thanks!

Reply:
@markwhitfeld one more thing about the `actions/checkout` action.

Sources should be checked out prior to any steps that use composite actions like `./github/actions/...`, because relative references requires already checkout out sources.
If you want to wrap the step that check out sources as a separate composite action, the only way to reference this composite action that will work, if I understand correctly, is `ngxs/store/.github/actions/...`

Reply:
Oh, hahaha. Yes, that makes total sense!